src/picongpu/include/traits/PICToOpenPMD.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2016 Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "traits/Unit.hpp"
+#include "traits/UnitDimension.hpp"
+
+#include "pmacc_types.hpp"
+#include "simulation_defines.hpp"
+
+#include <string>
+#include <vector>
+
+namespace picongpu
+{
+namespace traits
+{
+/** Reinterpret attributes for openPMD
+ *
+ * Currently, this conversion table is used to translate the PIConGPU
+ * globalCellIdx (unitless cell index) to the openPMD positionOffset (length)
+ */
+    template<typename T_Identifier>
+    struct OpenPMDName
+    {
+        std::string operator()() const
+        {
+            return T_Identifier::getName();
+        }
+    };
+
+    template<typename T_Type>
+    struct OpenPMDName<globalCellIdx<T_Type> >
```

Comment:
> please move the specializations to a *.unitless

this is an other abstraction between **PIConGPU** and **openPMD** and is therefore not in a `unitless` file (which converts between **user input** and **PIConGPU**).

We should set up a short offline meeting to make it more clear.

> Maybe your intention is that globalCellIdx is no user attributes, if so please create a *.tpp file in the same directory and add the specializations there. 

this might be better, let us talk offline to make sure we are talking about the same :)